src/include/exec_monitor.h

```diff
@@ -0,0 +1,15 @@
+#ifndef EXEC_MONITOR_H
+#define EXEC_MONITOR_H
+
+#include <linux/limits.h>
+
+struct process_info {
+	int ppid;
+	int pid;
+	int tgid;
+	char name[PATH_MAX];
+};
+
+void exec_monitor();
```

Comment:
Mhh I'm worried about this file. This currently serves two purposes:
It is used to declare the ABI between the kernel and the userspace, this needs to be strict C (for eBPF)
It is also used to declare the handy userspace function that will monitor execution events.

However one day your execution monitor will evolve to require C++ features and all of the sudden your eBPF program will stop compiling because of this header not being strict C anymore. This is because you currently expose exec_monitor to the eBPF program although it can not invoke it, this will lead to further  breakages.
Also you are exposing stuff such as struct process_info or PATH_MAX to any module which decides to call exec_monitor but as I already explained somewhere else you should keep your headers minimal to only cover the strict necessary. Config should never have to know about the process_info structure so you shouldn't expose it in the first place to keep the contact surface between your subsystems minimal.